cpp/CMakeLists.txt

```diff
@@ -69,8 +64,13 @@ option(SINGLEGPU "Disable all mnmg components and comms libraries" OFF)
 option(USE_CCACHE "Cache build artifacts with ccache" OFF)
 option(CUDA_STATIC_RUNTIME "Statically link the CUDA toolkit runtime and libraries" OFF)
 option(CUML_USE_RAFT_STATIC "Build and statically link the RAFT libraries" OFF)
+option(CUML_EXCLUDE_RAFT_FROM_ALL "Exclude RAFT targets from cuML's 'all' target" OFF)
```

Comment:
do we need individual `EXCLUDE_FROM_ALL` flags or an encompassing one would be goood enough? Even if it's not enough, it could be useful

Reply:
We don't need the individual ones, it just felt cleaner to do this on a per-library basis. If you prefer a single flag I'm happy to change it (note that we do need to make sure to forward the `RAFT_EXCLUDE_FAISS_FROM_ALL` option to the raft build appropriately).